cmake/modules/Finduuid.cmake

```diff
@@ -34,16 +34,18 @@ if(NOT UUID_INCLUDE_DIR)
   find_path(UUID_INCLUDE_DIR uuid/uuid.h)
 endif()
 
-if(EXISTS UUID_INCLUDE_DIR)
+if(EXISTS "${UUID_INCLUDE_DIR}")
   set(UUID_INCLUDE_DIRS ${UUID_INCLUDE_DIR})
   set(CMAKE_REQUIRED_INCLUDES ${UUID_INCLUDE_DIRS})
   check_cxx_symbol_exists("uuid_generate_random" "uuid/uuid.h" _uuid_header_only)
 endif()
 
-if(NOT _uuid_header_only AND NOT UUID_LIBRARY)
+if(NOT _uuid_header_only)
```

Comment:
Why did you remove `AND NOT UUID_LIBRARY`? This prevents the user from setting it to a particular version, as it will be overwritten.

Reply:
No, I set both. The problem in my opinion is that if the user has set `UUID_LIBRARY` then `UUID_LIBRARIES` was never set and that is the variable used to determine link dependencies

So `-DUUID_LIBRARY=some/lib.so` would not work because then `UUID_LIBRARIES` was not set. 

Now the logic is to only set `UUID_LIBRARIES` if the uuid is not only a header only library. And if the user did not set `UUID_LIBRARY ` its set to `uuid`, otherwise the user setting is propagated directly to `UUID_LIBRARIES`

Reply:
Yes, you are right. There is a bug in this file, but I think the solution is adding an `else` as per my comment below, so that when the user sets `UUID_LIBRARY`, it gets propagated to `UUID_LIBRARIES` directly. Otherwise this file will set it to `uuid` independently of what the user sets.